src/rust/engine/process_execution/src/local.rs

```diff
@@ -224,6 +208,19 @@ impl super::CommandRunner for CommandRunner {
             output_directory: snapshot.digest,
           })
           .to_boxed()
+      }).then(move |result| {
+        // Force workdir not to get dropped until after we've ingested the outputs
+        if !cleanup_local_dirs {
+          // This consumes the `TempDir` without deleting directory on the filesystem, meaning
+          // that the temporary directory will no longer be automatically deleted when dropped.
+          let preserved_path = workdir.into_path();
+          info!(
+            "preserved local process execution dir `{:?}` for {:?}",
+            preserved_path,
+            req_description
+          );
+        } // Else, workdir gets dropped here
```

Comment:
Not critical here, but in future, augmenting this comment with an explicit: `mem::drop(workdir)` wouldn't hurt.